### PR TITLE
[Merged by Bors] - feat: port Algebra.Ring.Divisibility

### PR DESCRIPTION
**Pull request by mpenciak**
Based off mathlib3 SHA: f1a2caaf51ef593799107fe9a8d5e411599f3996

### CODE REVIEW

**Comment by mpenciak** [comment]
Mathlib3port did a very good job on this file.

Only comment is about the `bit0`, `bit1` lemmas and whether they should be deleted outright or dealt with in another way.

File: Mathlib/Algebra/Ring/Divisibility.lean [hunk 1]
```diff
@@ -0,0 +1,158 @@
+/-
+Copyright (c) 2014 Jeremy Avigad. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jeremy Avigad, Leonardo de Moura, Floris van Doorn, Yury Kudryashov, Neil Strickland
+-/
```
Comment (semorrison) [inline Mathlib/Algebra/Ring/Divisibility.lean:5]:
```suggestion
Ported by: Matej Penciak
-/
```

File: Mathlib/Algebra/Ring/Divisibility.lean [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+variable [Add α] [Semigroup α]
+
+theorem dvd_add [LeftDistribClass α] {a b c : α} (h₁ : a ∣ b) (h₂ : a ∣ c) : a ∣ b + c :=
+  Dvd.elim h₁ fun d hd => Dvd.elim h₂ fun e he => Dvd.intro (d + e) (by simp [left_distrib, hd, he])
+#align dvd_add dvd_add
+
+end DistribSemigroup
+
+-- TODO : Delete this?
```
Comment (semorrison) [inline Mathlib/Algebra/Ring/Divisibility.lean:27]:
```suggestion
set_option linter.deprecated false in
```

**Comment by semorrison** [comment]
bors d+

**Pull request closed without merging**